File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
     ListNode ptr=head;
     ListNode pre=head;
     ListNode temp=head;
     if(head==null||head.next==null)
     return head;
     temp=temp.next;
     ListNode head1=temp;
     while(temp!=null&&temp.next!=null)   
     {
      pre.next=temp.next;
      pre=pre.next;
       temp.next=pre.next;
       temp=temp.next;
     }
     pre.next=head1;
     return head;
    }
}

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {

  ListNode *ptr=head;
  ListNode *pre=head;
   ListNode *en=head;
  if (head==NULL) return NULL;
  if(head->next==NULL)
  return head;
   while(en->next!=NULL)
   {
    en=en->next;
   }
   if(head->next->next==NULL)
   return head;
   ListNode *temp=en;
   while(ptr !=temp && ptr->next!=temp)
   {
    pre=pre->next;
    ptr->next=ptr->next->next;
    en->next=pre;
    en=en->next;
   // en->next=NULL;
    ptr=ptr->next;
    pre=ptr;

   }
   if(ptr->next==temp)
   {
   pre=pre->next;
   ptr->next=pre->next;
   en->next=pre;
    en=en->next;

    //pre->next=NULL;
   }
    en->next=NULL;
  return head;
        
    }
};